### DISCUSSION TITLE
Load option picker long list on Page load

### DISCUSSION THREAD

**Discussion by sghriyadh** [Budibase/budibase#11497] [Help] [answered]
Hi

i have nearly 20,000 options. Each option is 3-6 words. 

When the user clicks option picker then there is delay in loading the options as list is long. However, once loaded it works fine.

Is it possible to Load all the options when the page opens instead of loading after user clicks that option picker field?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @sghriyadh,

- When you say 20,000 options is this within an options column within your data area?

If its this one, there isn't any way currently to load these after page load. I would suggest creating a feature request https://github.com/Budibase/budibase/issues/new/choose which would hopefully get this functionality by the team.
- Or are these options loaded into a option's picker field through a different table via the data provider option?

If it is this option, you may be able to use this custom plugin; https://github.com/ConorWebb96/bb-type-ahead, this doesn't load the results on page load but instead loads results that match the search parameters. Making it super fast as it wouldn't be loading the 20,000 items.

I hope this makes sense and is helpful.

**Reply by sghriyadh** [reply]
Hi

Yes it is within a option column in my data area.

i have current form in one table with 5  fields. i have a database of 20,000 clients with their National id number date of birth mobile numbers. i am try to search client using their ID number. if ID found then i want the remaining parameters like date of jorth , mobile number to liad automatically. 

.I can create another table but setting a query using plugin i haven't done before. Is it like uploading plugin using GUI or i have to install plugin using command line? i am using windows 10 Docker engine.

**Reply by melohagan** [reply]
To add a comment on this one, having an options picker with 20,000 options isn't going to be fun for your end users 😁 

Seems like you would definitely want some searching / filtering to be added in this case, such as the type ahead component that Conor suggested.

Also for large numbers of options, instead of adding them directly into the *Options* type column, you can use a data table as your options source, using a data provider which could be filtered based on a text field for example: https://docs.budibase.com/docs/option-picker#using-a-data-provider

**Reply by sghriyadh** [reply]
@melohagan 

Thank you. Creating a separate table and dataprovider i understand.  

I want to clarify one thing,however.

![Screenshot_20230814_131454_Teams](https://github.com/Budibase/budibase/assets/130876667/06d6328c-ce25-4a7e-8581-ca812267106a)


As shown in the screenshot,  i have over 70,000 rows in a separate table.

 I have form which has following fields
1- File number
2- Name
3-Phone
i want to search the 70,000 for a specific file number.

if that file number found then i want to fill my form fields automatically with the Filenum er Name+phone.

Can it be done using budibase ?

**Reply by melohagan** [reply]
Yes! The update mode Form Block already includes the Data provider / repeater combo that has the filter on some ID.

You just need to add a search box and link that to the Data provider filter: https://docs.budibase.com/docs/adding-a-search-field#2-using-a-form-and-text-input-flexible-way

The only tricky part may be that the form values may only update on load of the screen. In this case you may want to either split your entry form and search into separate screens, or use conditional UI to only show the form after you have searched for an item.
